08-Vault.md

```diff
@@ -0,0 +1,39 @@
+# Vault
+
+* Constructor sets the password to a `private` variable 
+* Does `private` mean that value stored in the blockchain is not visible?
+
+<details>
+  <summary>EXPAND FOR SOLUTION</summary>
+  <p>
+  
+  Ethereum is a public ledger and it is possible to see all contract data sent on-chain. 
+  
+  Setting the `password` variable to private will only mean that internal calls from the very same contract can access it but that doesn't mean the value stored in that variable is not visible.
+
+  All we need to do here is just use the `web3.eth.getStorageAt()` function to inspect the value stored in the `password` variable.
+
+  Solution:
+
+  ```
+  var password = web3.eth.getStorageAt(instance, index);`
+  // For Metamask use callback function
+  web3.eth.getStorageAt(instance.address, 1, (err,res)=>{password=res});
```

Comment:
This may be an easier example.
```suggestion
  web3.eth.getStorageAt(contract.address, 1, function(error, result) {password = web3.toAscii(result)})
```